packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -53,6 +66,265 @@ This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aw
   - [Resource Servers](#resource-servers)
   - [Domains](#domains)
 
+## Identity Pools
+
+Identity pools provide temporary AWS credentials for users who are guests (unauthenticated) and for users who have been authenticated and received a token. An identity pool is a store of user identity data specific to your account.
+
+Identity pools can be used in conjunction with Cognito User Pools or by accessing external federated identity providers directly.
+
+
+### Authenticated and Unauthenticated Identities
+
+Identity pools define two types of identities: authenticated and unauthenticated. Every identity in your identity pool is either authenticated or unauthenticated. Authenticated identities belong to users who are authenticated by a public login provider (Amazon Cognito user pools, Login with Amazon, Sign in with Apple, Facebook, Google, SAML, or any OpenID Connect Providers) or a developer provider (your own backend authentication process). Unauthenticated identities typically belong to guest users.
+
+A basic Identity Pool with minimal configuration consists of two roles: one to apply to authenticated identities and one to apply to unauthenticated identities:
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  authenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-authenticated-role'),
+  unauthenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-unauthenticated-role'),
+});
+```
+
+
+### Federated Identity Providers
+
+You can associate identity providers with an Identity Pool by first associating them with a Cognito User Pool or by associating the provider directly with the identity pool.
+
+#### Associating a Provider Through a User Pool
+
+```ts
+const userpool = new cognito.UserPool(this, 'Pool');
+
+const provider = new cognito.UserPoolIdentityProviderAmazon(this, 'Amazon', {
+  clientId: 'amzn-client-id',
+  clientSecret: 'amzn-client-secret',
+  userPool: userpool,
+});
+
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  authenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-authenticated-role'),
+  unauthenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-unauthenticated-role'),
+  userPools: [userpool]
+});
+```
+
+When associating a user pool with an identity pool, a `UserPoolClient` is automatically created to handle the user pool's providers. If you want to control the the settings of the `UserPoolClient`, you can do that using `IdentityPoolProps.defaultClientOptions`:
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  authenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-authenticated-role'),
+  unauthenticatedRole: Role.fromRoleArn('arn:aws:iam::123456789012:role/my-unauthenticated-role'),
+  userPools: [userpool],
+  defaultClientOptions: {
```

Comment:
Given that we take a list of user pools, it's possible that a user may want to configure the client for each user pool differently.

Instead of accepting `userPools` and `defaultClientOptions`, it feels better instead to take a list of user pool clients directly. What do you think?

Reply:
Somehow i completely missed the existence of `UserPoolClientProps` which is essentially the `UserPoolClientOptions` plus an `IUserPool` to attach. An array of `userPoolProps` instead of the `userPools` and `defaultClientOptions` fields would make a lot more sense. I'll make that change.